File: viral_advertising/va.py
def viral_advertising(n):
    likes, shares = 2, 5
    for i in range(1, n):
        shares = (shares//2)*3
        likes += (shares//2)
    return likes

if __name__ == "__main__":
    print(viral_advertising(5))
    print(viral_advertising(3))

File: viral_advertising/va.js
const viralAdvertising = (n) => {
    let likes = 2, shares = 5;
    for(let i=1; i<n; i++){
        shares = ~~(shares/2) *3;
        likes += ~~(shares/2);
    }
    return likes;
}

console.log(viralAdvertising(5));
console.log(viralAdvertising(3));